sdk/helper/testhelpers/schema/response_validation.go

```diff
@@ -126,3 +126,22 @@ func GetResponseSchema(t *testing.T, path *framework.Path, operation logical.Ope
 
 	return &schemaResponses[0]
 }
+
+func ResponseValidatingCallback(t *testing.T) func(logical.Backend, *logical.Request, *logical.Response) {
+	return func(b logical.Backend, req *logical.Request, resp *logical.Response) {
+		if b == nil {
+			t.Fatalf("non-nil backend required")
+		}
+		backend, ok := b.(*framework.Backend)
+		if !ok {
+			t.Fatalf("could not cast %T to *framework.Backend", b)
+		}
+
+		ValidateResponse(
```

Comment:
@averche What happens if this backend calls an endpoint that doesn't have an established response structure defined already? Will this fail the test?

Reply:
I'm pretty sure it will fail here:

https://github.com/hashicorp/vault/blob/eef2390ec238d92543b58e075894eb51d7053097/sdk/helper/testhelpers/schema/response_validation.go#L119-L125

Reply:
So I'm guessing we would only want to add the callback for things that already implement response structures.

Reply:
I think this will be an issue because most test cases create one backend using this top level function that make a lot of calls that may not be related to the tested endpoint in order to set things up/ modify the test env.

Reply:
Yes, I had a similar concern but my guess is that as long as tests are specific enough, we can add the callback selectively for things that have already implemented the response structures. Since we will soon be covering every `/sys` request, I don't foresee many issues with this approach.

Reply:
We discusses out of band how this will be handled and is no longer a concern.